client/signup/config/steps-pure.js

```diff
@@ -117,8 +117,8 @@ export function generateSteps( {
 			stepName: 'plans-site-selected',
 			apiRequestFunction: addPlanToCart,
 			dependencies: [ 'siteSlug' ],
-			providesDependencies: [ 'cartItem', 'themeSlugWithRepo' ],
-			optionalDependencies: [ 'themeSlugWithRepo' ],
+			providesDependencies: [ 'cartItem', 'themeSlugWithRepo', 'addOnItem' ],
```

Comment:
**Note:**

We have to specify add-on items for so this many steps because we want to add a new dependency for `plans`, which has a number of linked steps https://github.com/Automattic/wp-calypso/blob/c33f1f6be415ba7adf63c2995eadd0cda4081e68/client/signup/config/step-components.js#L28-L42

More about bugs with linked steps and unspecified dependencies can be found here https://github.com/Automattic/wp-calypso/blob/c33f1f6be415ba7adf63c2995eadd0cda4081e68/client/signup/config/test/index.js#L23-L36